controllers/operands/kubevirtConsolePlugin_test.go

```diff
@@ -292,6 +292,59 @@ var _ = Describe("Kubevirt Console Plugin", func() {
 			Expect(hco.Status.RelatedObjects).To(Not(ContainElement(*objectRefOutdated)))
 			Expect(hco.Status.RelatedObjects).To(ContainElement(*objectRefFound))
 		})
+
+		Context("Node Placement", func() {
+			It("should add node placement if present in HCO CR", func() {
+				hco.Spec.Workloads.NodePlacement = commonTestUtils.NewNodePlacement()
+				hco.Spec.Infra.NodePlacement = commonTestUtils.NewOtherNodePlacement()
+
+				existingResource, err := NewKvUiPluginDeplymnt(hco)
+				Expect(err).ToNot(HaveOccurred())
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := newConsoleHandler(cl)
+				res := handler.ensure(req)
+				Expect(res.Created).To(BeFalse())
+				Expect(res.Updated).To(BeFalse())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Err).ToNot(HaveOccurred())
+
+				foundResource := &appsv1.Deployment{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).ToNot(HaveOccurred())
+
+				Expect(existingResource.Spec.Template.Spec.NodeSelector).ToNot(BeNil())
+				Expect(foundResource.Spec.Template.Spec.NodeSelector).ToNot(BeNil())
+				Expect(foundResource.Spec.Template.Spec.NodeSelector).To(Equal(existingResource.Spec.Template.Spec.NodeSelector))
+			})
+			It("should not add node placement if not present in HCO CR", func() {
```

Comment:
@tiraboschi I don't feel like this test adds much value, but I have added it anyway. Should I keep it or remove it?

Reply:
Yes, I think it adds value and probably we also miss other cases.
Please check https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/controllers/operands/ssp_test.go#LL133-L301 as a refrence.

Reply:
About the other test cases that are missing here, I'm certain that the one with `should overwrite node placement if directly set on` would be addressed after we take care of the reconciliation part. But I'm not 100% sure if `should modify node placement according to HCO CR` would also be taken care of after addressing reconciliation aspect (I'm inclined to think it is). WDYT?